torch_agents/cogment_verse_torch_agents/selfplay_td3/model.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2021 AI Redefined Inc. <dev+cogment@ai-r.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import random
+import time
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+# pylint: disable=R1725
+class ActorNetwork(nn.Module):
+    def __init__(self, **params):
+        """Initialize a ActorNetwork.
+        Params
+        ======
+            name (str): Agent name
+            obs_dim1 (int): number of features in agent's gpos
+            obs_dim2 (int): number of features in agent's goals
+            obs_dim3 (int): number of channels in agent's grid
+
+            number_actions (int): number of actions of the agent
+        """
+        super(ActorNetwork, self).__init__()
+        self.agent = params["name"]
+
+        if self.agent == "bob":
+            self.state_dim = params["obs_dim1"] + params["obs_dim2"]
+        else:
+            self.state_dim = params["obs_dim1"]
+
+        self.grid_channels = params["grid_shape"][2]
+
+        self.action_dim = params["act_dim"]
+        self.action_scale = torch.FloatTensor(params["action_scale"]).to(device)
+        self.action_bias = torch.FloatTensor(params["action_bias"]).to(device)
+        self.max_action = params["max_action"]
+
+        self.cnnl1 = nn.Sequential(
+            nn.Conv2d(self.grid_channels, 5, (4, 4)),
+            nn.LeakyReLU(),
+            nn.Conv2d(5, 10, (5, 5), stride=2),
```

Comment:
I think we should read these values from "params" or config. Ideally, I want to play around with (3,3) kernel sizes and stride=1